hub/models.py

```diff
@@ -3481,7 +3484,17 @@ def get_import_data(self):
             layer["source"] for layer in self.get_layers() if layer.get("visible", True)
         ]
         return GenericData.objects.filter(
-            data_type__data_set__external_data_source_id__in=visible_layer_ids
+            models.Q(data_type__data_set__external_data_source_id__in=visible_layer_ids)
+            & (
+                (
+                    models.Q(data__startswith="action_network")
+                    & (
+                        models.Q(json__email_addresses__0__status="subscribed")
+                        | models.Q(json__phone_numbers__0__status="subscribed")
+                    )
+                )
+                | (~models.Q(data__startswith="action_network"))
+            )
         )
```

Comment:
Another approach might be to [concat (`.union()`) the querysets](https://docs.djangoproject.com/en/4.2/ref/models/querysets/#django.db.models.query.QuerySet.union) for each individual layer.

This might be more resilient to future requirements such as per-layer custom filters.